### ISSUE TITLE
Problem with .snap files

### ISSUE THREAD

**Issue by aleksey-hoffman** [nklayman/vue-cli-plugin-electron-builder#1080]
**Describe the bug**
For some reason, I cannot install `.snap` files built with electron-builder:

```
yarn electron:build
sudo snap install --dangerous dist_electron/app-name_1.0.0_amd64.snap
```
I get the following error:

```
error: cannot install snap file: snap is unusable due to bad permissions; contact developer
```

I've never built / installed `.snap` files before, so I'm not sure if I'm doing it right or not.
I manually changed permissions of the snap file to `read & write` and selected `Allow executing file as program` and then tried to install it again, but I got the same error.

Then I ran `journalctl -u snapd` and here's the last line from it, not sure if that's what causes the problem:

```
loka 12 14:59:25 name-VirtualBox snapd[546]: container.go:199: in snap "app-name": "meta/gui/icon.png" should be world-readable, and isn'
```

Someone opened a similar issue in the electron-builder's repo: https://github.com/electron-userland/electron-builder/issues/5067
But my problem is I get the error trying to install the `.snap`, so even though it gets built successfully, clearly there's something wrong with the build config and I'm not sure if it's a problem with `vue-cli-plugin-electron-builder` or `electron-builder` or if it's something I'm doing wrong.

### vue.config.js
```
...
builderOptions: {
  icon: 'build/icon.png',
  snap: {
    confinement: 'devmode',
    grade: 'devel'
  },
  linux: {
    target: 'snap',
    icon: 'build/icon.png',
    category: 'system',
    maintainer: 'name'
  }
  ...
}
...
```
### Build log

```
 INFO  Building app with electron-builder:
  • electron-builder  version=21.2.0 os=5.4.0-48-generic
  • writing effective config  file=dist_electron/builder-effective-config.yaml
  • rebuilding native dependencies  dependencies=diskusage@1.1.3, xxhash-addon@1.3.0 platform=linux arch=x64
  • rebuilding native dependency  name=diskusage version=1.1.3
  • rebuilding native dependency  name=xxhash-addon version=1.3.0
  • packaging       platform=linux arch=x64 electron=10.1.3 appOutDir=dist_electron/linux-unpacked
  • building        target=snap arch=x64 file=dist_electron/app-name_1.0.0_amd64.snap
 DONE  Build complete!
```

**Environment:**
- OS: Ubuntu 20.04 x64
- electron: 10.1.3
- vue-cli-plugin-electron-builder: 1.4.6 (the issue occurs in the latest 2.0.0-rc.4 as well)

#### Relevant docs:
Snapcraft: https://snapcraft.io/docs/electron-apps
Electron-builder: https://www.electron.build/configuration/snap

**Comment by nklayman** [comment]
My guess is that the issue you linked on electron-builder is the culprit. Try cloning [the electron-builder quickstart repo](https://github.com/electron-userland/electron-webpack-quick-start), and building a snap with the config you posted. If that fails to install, then the issue is caused by electron builder. If not, I will see if there is something with this plugin that causes the issue.

**Comment by aleksey-hoffman** [comment]
@nklayman I just tried the quick-start boilerplate you suggested on the same Ubuntu 20.04, and it works perfectly, it built a `.snap` file by default without any config. And then I was able to install it with the same command I mentioned in the issue, without any errors.

To reproduce, you can install yarn and then:
```
cd electron-webpack-quick-start
rm -rf .git
yarn upgrade electron --latest
yarn compile
yarn dist
```

Then to install the `dist/electron-webpack-quick-start_0.0.0_amd64.snap` I just ran the following command:
```
sudo snap install --dangerous dist/electron-webpack-quick-start_0.0.0_amd64.snap
```
(`snap` is included in Ubuntu 20.04, no need to install it)

And it successfully installed the app without any errors (and I didn't even config anything). I can see it in the Ubuntu app launcher and can launch it without any problems.

The reason I created this issue is because `.snap` files are quite an important build target. They are not just "portable" executables like `.AppImage` files, they can actually be submitted to https://snapcraft.io/ and added to Ubuntu store, which is a great way to share your app with people in a more user-friendly way.

**Comment by nklayman** [comment]
Okay, thanks for testing that. I've been pretty busy lately but when I get the chance I'll test the snap target myself and see if I can figure out what is going wrong.

**Comment by nklayman** [comment]
Seems like the issue is coming from your `confinement` config. I removed that line and it worked fine for me. Try removing that and let me know if that fixes your issue.

**Comment by aleksey-hoffman** [comment]
@nklayman thanks for testing it, mate. I tried that. Turns out, in my case, the cause of the problem had nothing to do with the config properties themselves. For some unimaginable reason, the icon that I specified `build/icon.png` in the config had some read permissions denied (only on Ubuntu as well). 

I just changed all the file's permissions to `read & write`, and it got installed successfully after that.
